CDCgov/prime-reportstream
# Add Ingestion Time to the Deliveries Endpoint

This PR adds root report IDs and ingestion times to the Delivery API.

Test Steps:
1. Have a receiver with batching enabled.
2. Send a report into RS twice in a row.  The sender/receiver config should be such that these reports will be delivered to the aforementioned receiver.
3. Call the `/waters/report/{id}/delivery` API, using the send pipeline step reportId.
4. See the new `originalIngestion` field that includes all the possible root `reportId`s and their ingestion times (`createdAt`).

## Changes
- Added a new field to the `DeliveryHistory` class to contain the ingestion report IDs and times.
- Created a new `getRootReports` (notice the plurality) method, and added it to `ReportGraph` and `ReportService`.  These methods return a list of `ReportFile`s and work even when given a `reportId` from the batch or send step.
- Call the new `getRootReports` and extract the `reportId` and `createdAt` and put this information in the new `DeliveryHistory` field.
- Some kick ass unit tests.

## Checklist

### Testing

### Process

## Linked Issues

- Helps address https://github.com/CDCgov/trusted-intermediary/issues/989
